File: src/main/java/Entity/Evenement.java
package Entity;

import javafx.scene.control.Label;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;
import javafx.scene.layout.VBox;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Evenement {
    private int id_evenement;
    private String titre;
    private Date date;
    private int duree;
    private String lieu;
    private String objectif;
    private String image;
    private int id_actualite;
    private int id_user;

    public Evenement() {
    }

    public Evenement(String titre, Date date, int duree, String lieu, String objectif, String image, int id_actualite, int id_user) {
        this.titre = titre;
        this.date = date;
        this.duree = duree;
        this.lieu = lieu;
        this.objectif = objectif;
        this.image = image;
        this.id_actualite = id_actualite;
        this.id_user = id_user;
    }

    public int getId_user() {
        return id_user;
    }

    public Evenement(int id_evenement, String titre, Date date, int duree, String lieu, String objectif, String image, int id_actualite) {
        this.id_evenement = id_evenement;
        this.titre = titre;
        this.date = date;
        this.duree = duree;
        this.lieu = lieu;
        this.objectif = objectif;
        this.image = image;
        this.id_actualite = id_actualite;
    }

    public Evenement(String titre, Date date, int duree, String lieu, String objectif, String image, int id_actualite) {
        this.titre = titre;
        this.date = date;
        this.duree = duree;
        this.lieu = lieu;
        this.objectif = objectif;
        this.image = image;
        this.id_actualite = id_actualite;
    }
    public Evenement(String titre) {
        this.titre = titre;
    }

    public int getId_evenement() {
        return id_evenement;
    }

    public void setId_evenement(int id_evenement) {
        this.id_evenement = id_evenement;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public int getDuree() {
        return duree;
    }

    public void setDuree(int duree) {
        this.duree = duree;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public String getObjectif() {
        return objectif;
    }

    public void setObjectif(String objectif) {
        this.objectif = objectif;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getId_actualite() {
        return id_actualite;
    }

    public void setId_actualite(int id_actualite) {
        this.id_actualite = id_actualite;
    }

    @Override
    public String toString() {
        return "Evenement{" +
                "id_evenement=" + id_evenement +
                ", titre='" + titre + '\'' +
                ", date=" + date +
                ", duree=" + duree +
                ", lieu='" + lieu + '\'' +
                ", objectif='" + objectif + '\'' +
                ", image='" + image + '\'' +
                ", id_actualite=" + id_actualite +
                '}';
    }


}

File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use App\Repository\EvenementRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: EvenementRepository::class)]
class Evenement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id_evenement = null;

    #[ORM\Column(length: 55)]
    private ?string $Titre = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $Date = null;

    #[ORM\Column]
    private ?int $Duree = null;

    #[ORM\Column(length: 55)]
    private ?string $Lieu = null;

    #[ORM\Column(length: 255)]
    private ?string $Objectif = null;

    #[ORM\ManyToOne(targetEntity: Actualite::class, cascade: ['persist', 'remove'])]
    #[ORM\JoinColumn(name: 'actualite_id', referencedColumnName: 'id_actualite')]
    private ?Actualite $actualite = null;

    #[ORM\Column(length: 255)]
    private ?string $image = null;

    #[ORM\ManyToOne(inversedBy: 'evenements')]
    private ?User1 $User = null;

    public function getId(): ?int
    {
        return $this->id_evenement;
    }

    public function getTitre(): ?string
    {
        return $this->Titre;
    }

    public function setTitre(string $Titre): static
    {
        $this->Titre = $Titre;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->Date;
    }

    public function setDate(\DateTimeInterface $Date): static
    {
        $this->Date = $Date;

        return $this;
    }

    public function getDuree(): ?int
    {
        return $this->Duree;
    }

    public function setDuree(int $Duree): static
    {
        $this->Duree = $Duree;

        return $this;
    }

    public function getLieu(): ?string
    {
        return $this->Lieu;
    }

    public function setLieu(string $Lieu): static
    {
        $this->Lieu = $Lieu;

        return $this;
    }

    public function getObjectif(): ?string
    {
        return $this->Objectif;
    }

    public function setObjectif(string $Objectif): static
    {
        $this->Objectif = $Objectif;

        return $this;
    }

    public function getActualite(): ?Actualite
    {
        return $this->actualite;
    }

    public function setActualite(?Actualite $actualite): static
    {
        // unset the owning side of the relation if necessary
        if ($actualite === null && $this->actualite !== null) {
            $this->actualite->setEvenement(null);
        }

        // set the owning side of the relation if necessary
        if ($actualite !== null && $actualite->getEvenement() !== $this) {
            $actualite->setEvenement($this);
        }

        $this->actualite = $actualite;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): static
    {
        $this->image = $image;

        return $this;
    }

    public function getUser(): ?User1
    {
        return $this->User;
    }

    public function setUser(?User1 $User): static
    {
        $this->User = $User;

        return $this;
    }
}